### ISSUE TITLE
Fix EventSource ETW Session Logic (affects EventPipe)

### ISSUE THREAD

**Issue by vancem** [dotnet/coreclr#20915]
In EventProvider.cs there is code that code that shared between EventPipe and ETW in the method EtwEnableCallBack that tries to support Session-based filtering.     Unfortunately, in the ports from Desktop the Session based functionality was broken.  (see below), and in addition this code is causing grief for EventPipe (which shares much of the ETW code path), and has broken it already (we patched it once) and is likely to again.   

This issue tracks cleaning this up.   The recommendation is to remove the Session-based logic for ETW (it did not work anyway, and has not proven valuable so far).   As a second step (probably as a separate issue), we can decide to implement Session-based filtering for EventPipe (which is significantly easier), and if we ultimately care (we probably wont) we could do it properly for ETW (importantly, since the time we implemented this for ETW we have learned of better ways to doing it, so the current code is never useful (we would do it very differently, even for ETW).  

BACKGROUND:   First, what is Session-based filtering.

EventSource has a multiple-reader model.   Each reader is called a session and when sessions enable an EventSource the provide bitmask (keywords), and level (verbosity), which determine which events to give to the session.   Importantly, you can have multiple sessions (readers) connected simultaneously and they should not interfere with one another (each gets the events they asked for and are unaware of the other sessions).   Thus logically each session has the bitmask and verbosity for the EventSource associated with it and when an event is written, logically there is a loop over all sessions which writes the event to just those sessions that want it.   

This all works well.   The issue is when you have parameters for EventSource that is the problem.   This is not a theoretical issue.   Much of our instrumentation now is DiagnosticSource, and you turn on these events via a special EventSource.   You specify exactly which DiagnosticSource events you want by using EventSource parameters.   If you have more than two EventSources that want to listen to DiagnositicSource events, then you will have the issue I am about to describe.  

The issue is that it is a PARTICULAR EventSource (e.g. DiagnosticSourceEventSource), that knows how to interpret the string, but the only mechanism it has to write will write to ALL sessions unconditionally.   Thus for example if two sessions give filter parameters to DiagnosticSourceEventSource, turning on different events, then the second one 'win's and the first session does no get what it wants.   'Session based filtering fixes this  so that each session gets the events it asked for and the two sessions do not interfere).  

Fixing this is not actually trivial, because it requires a 'improved' contract between the EventSource base class and subclasses like DiagnosticSourceEventSource.   I will open a separate issue for that.

But suffice it to say that an attempt was made to do this for ETW (that is different ETW sessions would be distinguished) but it was tricky because unlike EventPipe or EventListener, our code cannot 'iterate' over the session (we see just the OS WriteEvent and the dispatch to individual sessions happens in the OS).    Suffice it to say, it added a bunch of complexity, and ultimately did not work.   

This issue tracks SIMPLY REMOVING THIS SUPPORT.  

The rational here is that we actually care more about EventPipe (which is cross platform) rather than ETW so the likely multi-session case for ETW is not likely.   Also it does not work today (so we are removing nothing, but does simplify things (in a non-trivial way).  

Basically we need to remove GetSessions() and all its related code (which includes SessionMask.   You can remove all references to perEventSourceSessionId and etwSessionId.  

FYI, if we ever decide we actually want to implement this per-session functionality on ETW, there is a .NET Framework bug 

See bug https://devdiv.visualstudio.com/DevDiv/_workitems/edit/105200 Using ETW session-based filtering with EventWriteEx() instead of custom reserved "Session" keywords

That goes into some of the details.     I don't expect us to do this any time soon.
